src/api/modelTypes.js

```diff
@@ -655,63 +674,76 @@ export type Submessage = $ReadOnly<{|
  * `Message`: `PmMessage` and `StreamMessage`.
  */
 type MessageBase = $ReadOnly<{|
-  /** Our own flag; if true, really type `Outbox`. */
-  isOutbox: false,
-
-  /**
-   * These don't appear in `message` events, but they appear in a `/message`
-   * response when a search is involved.
-   */
-  match_content?: string,
-  match_subject?: string,
-
-  /**
-   * The `flags` story is a bit complicated:
-   *  * Absent in `event.message` for a `message` event... but instead
-   *    (confusingly) there is an `event.flags`.
-   *  * Present under `EVENT_NEW_MESSAGE`.
-   *  * Present under `MESSAGE_FETCH_COMPLETE`, and in the server `/message`
-   *    response that action is based on.
-   *  * Absent in the Redux `state.messages`; we move the information to a
-   *    separate subtree `state.flags`.
-   */
-  flags?: $ReadOnlyArray<string>,
-
-  /**
-   * Present under EVENT_NEW_MESSAGE and under MESSAGE_FETCH_COMPLETE,
-   * and in `state.messages`, all as an AvatarURL, because we
-   * translate into that form at the edge.
-   *
-   * For how it appears at the edge (and how we translate) see
-   * AvatarURL.fromUserOrBotData.
-   */
-  avatar_url: AvatarURL,
```

Comment:
modelTypes [nfc]: Reorder MessageBase type's properties to follow the doc
    
    And give a clear section to the ones that aren't present in
    `message` events. (The jsdoc suggests we should primarily model the
    Message type on `message` events.)

Good idea!

`avatar_url` should stay at the top of the list, right? I see it at the top here: https://zulip.com/api/get-events#message

Reply:
(It has a different type in the actual event, but is present, and it's another form of the same data.)